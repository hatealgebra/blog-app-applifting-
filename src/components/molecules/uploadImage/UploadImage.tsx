import React from 'react';
import Button from '../../atoms/button/Button';
import { StyledMinimalButton, StyledUploadImage } from './uploadImage.styled';

const UploadImage = ({ image, setImage }: UploadImageProps) => {
  const [imagePreview, setImagePreview] = React.useState<string | undefined>(
    null
  );
  const hiddenFileInputRef = React.useRef<HTMLInputElement>(null);

  const handleInput = () => {
    if (hiddenFileInputRef.current !== null) {
      hiddenFileInputRef.current.click();
    }
  };

  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (event.target.files !== null) {
      const fileUploaded = event.target.files[0];
      setImage(fileUploaded);
    }
  };

  const resetImage = () => setImage(null);

  React.useEffect(() => {
    if (image !== null) {
      const objectURL = URL.createObjectURL(image);
      setImagePreview(objectURL);
    } else {
      setImagePreview(undefined);
    }
  }, [image]);

  return (
    <StyledUploadImage>
      <label className="upload-image__label">Featured Image</label>
      {image !== null ? (
        <>
          <img
            className="upload-image__preview"
            src={imagePreview}
            alt={image.name}
          />
          <div className="upload-image__button-row">
            <StyledMinimalButton onClick={handleInput}>
              Upload New
            </StyledMinimalButton>
            <StyledMinimalButton onClick={resetImage}>
              Delete
            </StyledMinimalButton>
          </div>
        </>
      ) : (
        <Button onClick={handleInput} colortheme="secondary">
          Upload an Image
        </Button>
      )}
      <input
        data-testid="image-uploader"
        ref={hiddenFileInputRef}
        type="file"
        style={{ display: 'none' }}
        accept="image/*"
        onChange={(e) => handleFileChange(e)}
      />
    </StyledUploadImage>
  );
};

export interface UploadImageProps {
  image: File | null;
  setImage: React.Dispatch<React.SetStateAction<File | null>>;
}

export default UploadImage;
