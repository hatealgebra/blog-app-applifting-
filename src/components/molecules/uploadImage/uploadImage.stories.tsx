import React from 'react';
import UploadImage from './UploadImage';

export const UploadImageExample = () => {
  const [image, setImage] = React.useState<File | null>(null);
  return <UploadImage image={image} setImage={setImage} />;
};

export const UploadImageWithFile = () => {
  const file = new File(['hello'], 'hello.png', { type: 'image/png' });
  const [image, setImage] = React.useState<File | null>(file);
  return <UploadImage image={image} setImage={setImage} />;
};
export default {
  title: 'Molecules/Upload image',
  component: UploadImage,
};
