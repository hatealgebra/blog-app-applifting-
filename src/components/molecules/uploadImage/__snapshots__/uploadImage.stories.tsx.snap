// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/Upload image UploadImageExample smoke-test 1`] = `
<div class="sc-ezGUZh fpAwaX">
  <label class="upload-image__label">
    Featured Image
  </label>
  <button variant="standard"
          colortheme="primary"
          type="button"
          class="sc-dkjKgF iJlhGR"
  >
    Upload an Image
  </button>
  <input data-testid="image-uploader"
         type="file"
         accept="image/*"
         style="display: none;"
  >
</div>
`;

exports[`Molecules/Upload image UploadImageWithFile smoke-test 1`] = `
<div class="sc-ezGUZh fpAwaX">
  <label class="upload-image__label">
    Featured Image
  </label>
  <img class="upload-image__preview"
       src="blob:http://127.0.0.1:6006/c8b9e969-b3eb-4a77-99d3-783eb690809f"
       alt="hello.png"
  >
  <div class="upload-image__button-row">
    <button variant="outline"
            colortheme="primary"
            type="button"
            class="sc-dkjKgF hhKsHh"
    >
      Upload New
    </button>
    <button variant="outline"
            colortheme="primary"
            type="button"
            class="sc-dkjKgF hhKsHh"
    >
      Delete
    </button>
  </div>
  <input data-testid="image-uploader"
         type="file"
         accept="image/*"
         style="display: none;"
  >
</div>
`;
